src/components/Post.tsx

```diff
@@ -0,0 +1,244 @@
+import createImageInfo from '../assets/home-mock-data';
+import type {PostType} from '../assets/posts-mock-data';
+import Animated, {
+  Easing,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+
+interface PostProps {
+  item: PostType;
+}
+
+export default function Post({item}: PostProps) {
+  const {height} = useWindowDimensions();
+  const maxHeroHeight = height - (64 + 60 + 40 + 60 + 10); // right under the header
+  const [isLiked, setIsLiked] = useState(false);
+  const [isShowMore, setIsShowMore] = useState(false);
+  const [contentHeight, setContentHeight] = useState(0);
+
+  const toggleShowMore = () => {
+    setIsShowMore(!isShowMore);
+  };
+
+  const textHeight = item.description ? 42 : 0;
+  const animatedHeight = useSharedValue(textHeight);
+  const animatedOpacity = useSharedValue(0);
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const height = isShowMore
+      ? Math.min(contentHeight, maxHeroHeight)
+      : textHeight;
+    return {
+      height: withTiming(height, {
+        duration: 300,
+        easing: Easing.inOut(Easing.ease),
+      }),
+    };
+  });
+
+  const handleAnimation = () => {
+    animatedOpacity.value = withTiming(isShowMore ? 0.8 : 0, {
+      duration: 300,
+      easing: Easing.inOut(Easing.ease),
+    });
+  };
+
+  useEffect(() => {
+    animatedHeight.value = isShowMore
+      ? Math.min(contentHeight, maxHeroHeight)
+      : textHeight;
+    handleAnimation();
+  }, [isShowMore, contentHeight]);
+
+  const onTextLayout = (event: LayoutChangeEvent) => {
+    setContentHeight(event.nativeEvent.layout.height);
+  };
+
+  const onContainerPress = () => {
+    if (isShowMore) {
+      toggleShowMore();
+    }
+  };
+
+  const navigation: any = useNavigation();
+  const ImageInfo = createImageInfo(navigation, isLiked, setIsLiked);
+
+  const DescriptionView = () => {
+    if (!item.description) {
+      return null;
+    }
+    if (item.description.length > 120) {
+      return (
+        <ScrollView
+          showsVerticalScrollIndicator={false}
+          style={{flexGrow: 0}}
+          contentContainerStyle={{
+            flexGrow: 0,
+          }}>
+          <Pressable onPress={toggleShowMore}>
+            <Text
+              style={styles.description}
+              onLayout={isShowMore ? onTextLayout : undefined}>
+              {isShowMore
+                ? item.description
+                : `${item.description.slice(0, 120)}...`}
+              <Text style={styles.showText}>
+                {isShowMore ? ' Show Less' : ' Show More'}
+              </Text>
+            </Text>
+          </Pressable>
+        </ScrollView>
+      );
+    } else {
+      return (
+        <Text style={styles.description} onLayout={onTextLayout}>
+          {item.description}
+        </Text>
+      );
+    }
+  };
+
+  return (
+    <Pressable onPress={onContainerPress} style={{height, flex: 1}}>
+      <LinearGradient
```

Comment:
move LinearGradient to a function/separate component and pass the style as a prop, as you use it with the same colors but the style is different